hbase-server/src/test/java/org/apache/hadoop/hbase/master/assignment/TestDeadServerMetricRegionChore.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.assignment;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseClassTestRule;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.testclassification.LargeTests;
+import org.apache.hadoop.hbase.testclassification.MasterTests;
+import org.apache.hadoop.hbase.util.FSUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * Base class for AM test.
+ */
+@Category({ MasterTests.class, LargeTests.class })
+public class TestDeadServerMetricRegionChore {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestDeadServerMetricRegionChore.class);
+
+  protected HBaseTestingUtility util;
+
+  protected void setupConfiguration(Configuration conf) throws Exception {
+    FSUtils.setRootDir(conf, util.getDataTestDir());
+    // Disable DeadServerMetricRegionChore
+    conf.setInt(AssignmentManager.DEAD_REGION_METRIC_CHORE_INTERVAL_MSEC_CONF_KEY, -1);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    util = new HBaseTestingUtility();
+    setupConfiguration(util.getConfiguration());
+  }
+
+  @After
+  public void tearDown() throws Exception {
+  }
+
+  @Test
+  public void testDeadServerMetricRegionChore() throws Exception {
+    try {
+      this.util.startMiniCluster();
+      // Master start up normal.
+      final TableName tableName = TableName.valueOf("testDeadServerMetricRegionChore");
+      this.util.createTable(tableName, "f");
+      this.util.waitTableAvailable(tableName);
+    } finally {
+      this.util.killMiniHBaseCluster();
```

Comment:
Move to tearDown()?

Reply:
I think it's better to have a shutdown, instead of kill the cluster, in `tearDown`. Also seems that an assertion in this test is missing.

Reply:
The test is failed when startMiniCluster
{code}
[INFO] Running org.apache.hadoop.hbase.master.assignment.TestDeadServerMetricRegionChore
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 210.246 s <<< FAILURE! - in org.apache.hadoop.hbase.master.assignment.TestDeadServerMetricRegionChore
[ERROR] org.apache.hadoop.hbase.master.assignment.TestDeadServerMetricRegionChore.testDeadServerMetricRegionChore  Time elapsed: 210.054 s  <<< ERROR!
java.io.IOException: Shutting down
	at org.apache.hadoop.hbase.master.assignment.TestDeadServerMetricRegionChore.testDeadServerMetricRegionChore(TestDeadServerMetricRegionChore.java:60)
Caused by: java.lang.RuntimeException: Master not initialized after 200000ms
	at org.apache.hadoop.hbase.master.assignment.TestDeadServerMetricRegionChore.testDeadServerMetricRegionChore(TestDeadServerMetricRegionChore.java:60)
{code}

Reply:
Is the shutdown failing or because it is part of the `tearDown`?

Reply:
The shutdown has no problem.